src/cider/exec/operator/join/HashTableFactory.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "cider/CiderException.h"
+
+namespace cider_hashtable {
+template <typename Key,
+          typename Value,
+          typename Hash,
+          typename KeyEqual,
+          typename Grower,
+          typename Allocator>
+template <typename... Args>
+std::unique_ptr<BaseHashTable<Key, Value, Hash, KeyEqual, Grower, Allocator>>
+HashTableSelector<Key, Value, Hash, KeyEqual, Grower, Allocator>::createForJoin(
+    HashTableType hashtable_type,
+    Args&&... args) {
+  switch (hashtable_type) {
+    case LINEAR_PROBING:
+      return std::make_unique<LinearProbeHashTable<Key, Value, Hash, KeyEqual>>(
+          std::forward<Args>(args)...);
```

Comment:
I'm confused with that `args`. If it stands for arguments in `Strategy2`. we don't need passing it down to HashTable side right? If it's for HashTable use, could you add some comments and examples to explain what may be in `args`?

Reply:
args are the arguments to hashtable impl's constructor. It can be zero or many

Reply:
So it's not the `args` in `Strategy2` for HashTable selection right?
It seems hard to use it since it is template, could you give an example for that?

Reply:
Ok, I will a example in next pr.